Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLReduction.py

```diff
@@ -533,7 +536,9 @@ def PyExec(self):
         ws = '__' + self.getPropertyValue('OutputWorkspace')
         # we do not want the summing done by LoadAndMerge since it will be pair-wise and slow
         # instead we load and list, and merge once with merge runs
-        LoadAndMerge(Filename=self.getPropertyValue('Run').replace('+',','), LoaderName='LoadILLSANS', OutputWorkspace=ws)
+        if not mtd.doesExist(ws):
```

Comment:
I think this condition should be removed, the caching is handled by the need_processing and need_loading, here it should always load.